keras/utils/vis_utils.py

```diff
@@ -13,7 +13,10 @@
 
 
 def _check_pydot():
-    if not (pydot and pydot.find_graphviz()):
+    try:
+        # Attempt to create an image of a blank graph to check the pydot/graphviz installation.
+        pydot.Dot.create(pydot.Dot())
+    except:
```

Comment:
Actually, never mind what I said, it will be safer to just catch `Exception`. Do add a comment on why we do this.